### DISCUSSION TITLE
Join event

### DISCUSSION THREAD

**Discussion by ImTheLeviDR** [Minestom/Minestom#2250] [Help]
I'm new. Anyone can help me making a join event?
I have this, but this isn't join. It's spawn which triggers when the player respawns.
`MinecraftServer.getGlobalEventHandler().addListener(PlayerSpawnEvent.class, event -> {

}`

**Comment by mudkipdev** [comment]
You can check [`isFirstSpawn`](https://javadoc.minestom.net/net/minestom/server/event/player/PlayerSpawnEvent.html#isFirstSpawn()), but if you want an event that's called before the player loads into a world, you can try [`AsyncPlayerConfigurationEvent`](https://javadoc.minestom.net/net/minestom/server/event/player/AsyncPlayerConfigurationEvent.html) or [`AsyncPlayerPreLoginEvent`](https://javadoc.minestom.net/net/minestom/server/event/player/AsyncPlayerPreLoginEvent.html).
